view/frontend/web/js/paypal_express/button.js

```diff
@@ -0,0 +1,654 @@
+                    }.bind(this));
+                } else {
+                    // Initialize the PayPal component if config is set
+                    if (!isConfigSet(paypalPaymentMethod, ['merchantId'])) {
+                        return;
+                    }
+                    this.initialisePaypalComponent(paypalPaymentMethod, element);
+                }
+            } else {
+                // Initialize PayPal component on product view page
+                this.initialiseonPDP(config, element);
+            }
+        },
+
+        initialiseonPDP: async function (config, element) {
+            // Configuration setup
+            try {
+                const response = await getExpressMethods().getRequest(element);
+                const cart = customerData.get('cart');
+                virtualQuoteModel().setIsVirtual(true, response);
+
+                cart.subscribe(function () {
+                    this.reloadPaypalButton(element);
+                }.bind(this));
+
+                setExpressMethods(response);
+                totalsModel().setTotal(response.totals.grand_total);
+                currencyModel().setCurrency(response.totals.quote_currency_code);
+
+                const $priceBox = getPdpPriceBox();
+                const pdpForm = getPdpForm(element);
+
+                $priceBox.on('priceUpdated', async function () {
+                    const isValid = new Promise((resolve, reject) => {
+                        return validatePdpForm(resolve, reject, pdpForm, true);
+                    });
+
+                    isValid
+                        .then(async function () {
+                            this.reloadPaypalButton(element);
+                        }.bind(this))
+                        .catch(function (error) {
+                            console.log(error);
+                        });
+                }.bind(this));
+
+                let paypalPaymentMethod = await getPaymentMethod('paypal', this.isProductView);
+
+                if (!paypalPaymentMethod) {
+                    console.error('PayPal payment method not found');
+                    return;
+                }
+
+                if (!isConfigSet(paypalPaymentMethod, ['gatewayMerchantId', 'merchantId'])) {
+                }
+                paypalPaymentMethod.configuration.intent = 'authorize';
+
+                this.initialisePaypalComponent(paypalPaymentMethod, element);
+            } catch (error) {
+                console.error('Error in initialiseonPDP:', error);
+            }
+        },
+
+
+        initialisePaypalComponent: async function (paypalPaymentMethod, element) {
+            // Configuration setup
+            const config = configModel().getConfig();
+            const adyenData = window.adyenData;
+            const adyenCheckoutComponent = await new AdyenCheckout({
+                locale: config.locale,
+                originKey: config.originkey,
```

Comment:
Is the originKey here still needed if the clientKey is being provided below?

Reply:
Yup, verified and we do not need it anymore
https://docs.adyen.com/development-resources/client-side-authentication/migrate-from-origin-key-to-client-key/